### PR TITLE
WIP: Add augmentors to chip classification

### PR DESCRIPTION
**Pull request by lmbak**
## Overview

PR that adds the most relevant augmentors from [albumentations ](https://github.com/albu/albumentations)to the chip classification task.

### Checklist

- [ ] Updated `docs/changelog.rst`
- [ ] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [ ] Ran scripts/format_code and committed any changes
- [ ] Documentation updated if needed
- [ ] PR has a name that won't get you publicly shamed for vagueness

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as rebuilding the Docker image.
* Include test case, and expected output if not captured by automated tests.

Closes https://github.com/azavea/raster-vision/issues/831

### CODE REVIEW

**Comment by lmbak** [comment]
@lewfish Finished most of the code related to the augmentors. It succesfully passes the unit_tests, but not the integration_tests. It hangs [here](https://github.com/lmbak/raster-vision/blob/lmb/augmentation/rastervision/backend/pytorch_chip_classification.py#208). Is that because it is still WIP, or did I miss anything?

File: requirements.txt [hunk 1]
```diff
@@ -15,3 +15,4 @@ rasterio==1.0.7
 protobuf==3.6.*
 scipy==1.1.*
 supermercado==0.0.*
+albumentations==0.3.3
```
Comment (lewfish) [inline requirements.txt]: This should be moved to the pytorch section in the `extra_requirements.json` file since it only applies to those backends.

File: rastervision/backend/torch_utils/chip_classification/data.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    train_dir = join(data_dir, 'train')
+    valid_dir = join(data_dir, 'valid')
+
+    aug_transform = [ToTensor()]
+    transform = [ToTensor()]
+
+    for augmentor in augmentors:
+        if augmentor == 'RandomHorizontalFlip':
+            aug_transform.append(HorizontalFlip(p=0.5))
```
Comment (lewfish) [inline rastervision/backend/torch_utils/chip_classification/data.py]: I thought the ToTensor() needs to come at the end of the list of transforms but I could be wrong.

Comment (lmbak) [reply rastervision/backend/torch_utils/chip_classification/data.py]: I think you are right. I have changed it. I'll test the debug chips tomorrow.

**Comment by lmbak** [comment]
@lewfish As discussed here the update. From now on I will most likely not be working on the subject for 3/4 weeks. As stated here https://github.com/azavea/raster-vision/issues/831 the last issue I was facing had to do with the format of the images (PIL.image vs numpy array). I have not gotten to the bottom of it yet, but it seems that albumentations expects a numpy array, whereas the [databunch ](https://github.com/lmbak/raster-vision/blob/lmb/augmentation/rastervision/backend/torch_utils/data.py )now contain [ImageFolder ](https://github.com/lmbak/raster-vision/blob/lmb/augmentation/rastervision/backend/torch_utils/chip_classification/data.py#L78) instances with PIL images inside them. Sorry for the mess of commits. I'll have to work on being more clean on those.

**Comment by lewfish** [comment]
> Sorry for the mess of commits. I'll have to work on being more clean on those.

That's ok. You might want to learn about interactive rebasing using git which lets you combine commits together.

**Comment by lewfish** [comment]
I'll probably continue with the work you did. Thanks!

**Comment by lewfish** [comment]
Welcome back. This PR is against an old branch that was closed, so I would suggest that you pull the latest `master` branch, make a branch off of it, and then `git cherry-pick` your commits onto it.

**Comment by lmbak** [comment]
Thanks! I'll get working on it.
